pkg/controllers/scheduler/util_test.go

```diff
@@ -112,3 +120,283 @@ func TestMatchedPolicyKey(t *testing.T) {
 		})
 	}
 }
+
+func generateFedObj(workload *unstructured.Unstructured) *fedcorev1a1.FederatedObject {
+	rawTargetTemplate, _ := workload.MarshalJSON()
+	return &fedcorev1a1.FederatedObject{
+		ObjectMeta: metav1.ObjectMeta{
+			Generation: 1,
+		},
+		Spec: fedcorev1a1.GenericFederatedObjectSpec{
+			Template: apiextensionsv1.JSON{Raw: rawTargetTemplate},
+		},
+	}
+}
+
+func Test_validatemigrationConfig(t *testing.T) {
+	sourceObject := &unstructured.Unstructured{
```

Comment:
NIT: would prefer these test specs to be organized as a table with a suitable name for each.

Reply:
Updated.